### PR TITLE
first try

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,8 +1,59 @@
...
+                map.replace(operationType, map.get(operationType) + amount);
+            }
+            map.putIfAbsent(operationType, amount);
+        }
+        int result = map.get("supply") - map.get("buy");
+        for (String key : map.keySet()) {
+            builder.append(key)
+                    .append(",")
+                    .append(map.get(key))
+                    .append(System.lineSeparator());
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:54]: remove empty line

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,8 +1,59 @@
...
+import java.util.Map;
+import java.util.TreeMap;
+
 public class WorkWithFile {
 
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> data = readFromFile(fromFileName);
+        String report = createReport(data);
+        writeToFile(toFileName, report);
+
```
Comment (zachar-chorny) [inline src/main/java/core/basesyntax/WorkWithFile.java:17]: empty line

**Review by zachar-chorny — CHANGES_REQUESTED** [review]
Good job. But let's add constants for all the magic numbers and strings

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,8 +1,60 @@
...
+        String report = createReport(data);
+        writeToFile(toFileName, report);
+    }
+
+    public List<String> readFromFile(String fromFileName) {
+        Path filePath = Paths.get(fromFileName);
+        try {
+            return Files.readAllLines(filePath);
+        } catch (IOException a) {
+            throw new RuntimeException("Something gone wrong to read the file");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:25]:
```suggestion
            throw new RuntimeException("Something gone wrong to read the file: " + fromFileName);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,8 +1,60 @@
...
+        }
+    }
+
+    public void writeToFile(String toFileName, String report) {
+        Path filePath = Paths.get(toFileName);
+        try {
+            Files.writeString(filePath, report);
+        } catch (IOException e) {
+            throw new RuntimeException("Something gone wrong to write to the file");
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:34]:
```suggestion
            throw new RuntimeException("Something gone wrong to write to the file: " + toFileName);
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,8 +1,60 @@
...
+            String operationType = split[0];
+            if (map.containsKey(operationType)) {
+                map.replace(operationType, map.get(operationType) + amount);
+            }
+            map.putIfAbsent(operationType, amount);
+        }
+        int result = map.get("supply") - map.get("buy");
+        for (String key : map.keySet()) {
+            builder.append(key)
+                    .append(",")
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/WorkWithFile.java:53]: Not fixed
image

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by kateryna-deshchinska — CHANGES_REQUESTED** [review]
You need to create constants for all magic number and string in your code, for example :
    private static String NEW_LINE = System.lineSeparator();
    private static int OPERATION_INDEX = 0;
    private static int AMOUNT_INDEX = 1;
    and the same for other string
    
Its are class fields, not inside of methods. And after that you may use this constant inside methods.

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,8 +1,75 @@
...
+                map.replace(operationType, map.get(operationType) + amount);
+            }
+            map.putIfAbsent(operationType, amount);
+        }
+        int supplyAmount = map.get("supply");
+        int buyAmount = map.get("buy");
+        int resultAmount = supplyAmount - buyAmount;
+        for (String operation : map.keySet()) {
+            stringBuilder.append(operation)
+                    .append(",")
```
Comment (kateryna-deshchinska) [inline src/main/java/core/basesyntax/WorkWithFile.java:58]: also, make constant for coma

**Review by kateryna-deshchinska — APPROVED** [review]

**Pull request closed without merging**